src/client.ts

```diff
@@ -16,105 +21,175 @@ export type ResourceIteratee<T> = (each: T) => Promise<void> | void;
-    const request = new Promise<void>((resolve, reject) => {
-      http.get(
-        {
-          hostname: 'localhost',
-          port: 443,
-          path: '/api/v1/some/endpoint?limit=1',
-          agent: false,
-          timeout: 10,
-        },
-        (res) => {
-          if (res.statusCode !== 200) {
-            reject(new Error('Provider authentication failed'));
-          } else {
-            resolve();
-          }
-        },
-      );
+  constructor(readonly config: IntegrationConfig) {
+    this.axiosInstance = Axios.create({
+      headers: {
+        Authorization: 'Bearer ' + config.clientToken,
+      },
     });
+    this.sentryBaseUrl = 'https://sentry.io/api/0/';
+
+    this.sentryOrganization = config.organization;
+  }
 
+  public async verifyAuthentication(): Promise<void> {
+    // A call to the baseURL will return a valid 200 status as long as we have a
+    // valid bearer token for authentication.
     try {
-      await request;
+      await this.axiosInstance.get(this.sentryBaseUrl);
     } catch (err) {
+      const response = err.response || {};
+
       throw new IntegrationProviderAuthenticationError({
-        cause: err,
-        endpoint: 'https://localhost/api/v1/some/endpoint?limit=1',
-        status: err.status,
-        statusText: err.statusText,
+        endpoint: this.sentryBaseUrl,
+        status: response.status,
+        statusText: response.statusText,
       });
     }
   }
 
+  /**
+   * Iterates each group resource in the provider.
+   *
+   * @param iteratee receives each resource to produce entities/relationships
+   */
+  public async iterateOrganizations(
+    iteratee: ResourceIteratee<SentryOrganization>,
+  ): Promise<void> {
+    let url = this.sentryBaseUrl + 'organizations/';
+    let moreData = true;
+
+    //optionally add org_slug if it was provided in .env
+    if (this.sentryOrganization) {
+      url += `${this.sentryOrganization}/`;
+    }
+
+    while (moreData) {
+      const orgResponse = await this.axiosInstance.get(url);
+      const orgResults = orgResponse.data;
+
+      const orgHeaders = orgResponse.headers;
+      moreData = Boolean(orgHeaders.results); //results=true when more than 100 results are available
+
+      // Organization is an oddity in that it may at times only return a single object
+      // instead of an iterable array.  Check for an iterator before proceeding.
+      if (orgResults && typeof orgResults[Symbol.iterator] === 'function') {
+        for (const organization of orgResults) {
+          await iteratee(organization);
+        }
+      } else {
+        await iteratee(orgResults);
+      }
+    }
+  }
+
+  /**
+   * Iterates each group resource in the provider.
+   *
+   * @param iteratee receives each resource to produce entities/relationships
+   * @param orgSlug added to URL to specify correct Sentry organization
+   */
+  public async iterateTeams(
+    iteratee: ResourceIteratee<SentryTeam>,
+    organizationSlug: string,
```

Comment:
nit: I prefer putting the iteratee _last_ in these types of methods, just for readablity's sake.
```suggestion
  public async iterateTeamsForOrganization(
    organizationSlug: string,
    iteratee: ResourceIteratee<SentryTeam>,
```

then in the caller...
```ts
await client.iterateTeamsForOrganization(orgId, async (team) => { 
  // do something here 
});
```

Reply:
Planning to update as part of the next branch for INT-2067.

Reply:
Went ahead and updated in this branch.  Should be complete.